packages/data-stores/src/queries/use-has-3rd-party-cookies.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+
+let result: boolean;
+
+export function useHas3PC() {
+	const [ hasCookies, setHasCookies ] = useState( Boolean( result ) );
+
+	useEffect( () => {
+		const iframe = document.createElement( 'iframe' );
+
+		function handler( event: MessageEvent ) {
+			const { data } = event;
+			if ( data.type === 'widgets.wp.com-cookie-check' ) {
+				setHasCookies( data.result );
+				window.removeEventListener( 'message', handler );
+				iframe.remove();
+			}
+		}
+		if ( result === undefined ) {
+			iframe.src = 'https://widgets.wp.com/calypso-happychat/check-cookies.html';
+			iframe.style.display = 'none';
+
+			window.addEventListener( 'message', handler );
+			document.body.appendChild( iframe );
+			() => {
```

Comment:
Also, did we mean to actually `return` this cleanup callback?

Reply:
Yes, I think you're right 👍